src/main/java/io/javalin/core/compression/DynamicCompressionHandler.kt

```diff
@@ -0,0 +1,63 @@
+package io.javalin.core.compression
+
+import io.javalin.core.JavalinConfig
+import io.javalin.core.util.Header
+import io.javalin.http.Context
+import javax.servlet.http.HttpServletResponse
+
+/**
+ * Class designed to handle dynamic content compression for Javalin.
+ *
+ * Uses DynamicCompressionStrategy within JavalinConfig to determine the type and level of compression to be applied.
+ *
+ * @see DynamicCompressionStrategy
+ */
+class DynamicCompressionHandler(val ctx: Context, val config: JavalinConfig) {
+
+    val compressionStrategy = config.inner.dynamicCompressionStrategy
+
+    /**
+     * @param res The response that we wish to encode
+     */
+    fun compressResponse(res: HttpServletResponse) {
+        val resultStream = ctx.resultStream()!!
+        if (brotliShouldBeDone(ctx)) { //Do Brotli
+            val level = compressionStrategy?.brotliLevel ?: DynamicCompressionStrategy.BROTLI_DEFAULT_LEVEL
```

Comment:
This is what I commented on in the beginning of the review, if you set the default already in the config, you'll just have `val level = compressionStrategy.brotliLevel` here.